### ISSUE TITLE
convert ZXDiagram to tensor network

### ISSUE THREAD

**Issue by Roger-luo** [QuantumBFS/ZXCalculus.jl#16]

**Comment by GiggleLiu** [comment]
Might be useful in the future.
```
function ztensor(nleg::Int, α::T) where T<:Number
    shape = (fill(2, nleg)...,)
    factor = exp(im*α)
    out = zeros(typeof(factor), shape...)
    out[1] = one(typeof(factor))
    out[fill(2, nleg)...] = factor
    out
end

function xtensor(nleg::Int, α::T) where T<:Number
    shape = (fill(2, nleg)...,)
    reshape(reduce(kron, fill(pos, nleg)) + exp(im*α)*reduce(kron, fill(neg, nleg)), shape)
end

function htensor(nleg::Int, α::T) where T<:Number
    shape = (fill(2, nleg)...,)
    factor = exp(im*α)
    out = fill(one(typeof(factor)), shape)
    out[fill(2, nleg)...] = factor
    return out
end
```

**Comment by GiggleLiu** [comment]
## The correspondence with Yao.jl
```julia
θ = 0.5
@test mat(Rz(θ) * phase(θ/2)) ≈ ztensor(2, θ)
@test mat(Rx(θ) * phase(θ/2)) ≈ xtensor(2, θ)
@test mat(H) .* sqrt(2) ≈ htensor(2, π)
```

**Comment by exAClior** [comment]
This is implemented in #93 albeit for zxw diagram. I will close this for now.
